docs/verifiable-document/document-template.md

```diff
@@ -4,8 +4,228 @@ title: Deploying Document Renderer
+
+This is a live preview where you can see the changes when you:
+
+1. edit the unwrapped document data in the "Knobs" tab
+1. edit the template code to render the data
+
+## Developing the renderer
+
+Now that we have set up the development environment, we can start writing our renderer. We will first define the data structure of our Certificate of Completion, followed by writing the renderer to render the HTML code corresponding to the data provided.
+
+### Update sample document data and type
+
+To update the unwrapped document data and the corresponding data type, you will need to update the data definition file in `src/templates/customTemplate/samples.ts`:
+
+```js
+import { Document } from "@govtechsg/decentralized-renderer-react-components";
+
+export interface CustomTemplateCertificate extends Document {
+  name: string;
+  recipient: {
+    name: string
+  };
+}
+
+export const customTemplateCertificate: CustomTemplateCertificate = {
+  name: "OpenAttestation Tutorial Certificate of Completion",
+  recipient: {
+    name: "John Doe"
+  },
+  $template: {
+    name: "COC",
+    type: "EMBEDDED_RENDERER",
+    url: "http://localhost:3000"
+  }
+};
+```
+
+In the above document data, you see three root objects:
+
+### 1. \$template
+
+The `$template` key to describe the template name used to render this display.
+
+`$template.type` will always take the value of `EMBEDDED_RENDERER` for documents rendered in this manner.
+
+`$template.name` is the name of the template used to render a given OA document. This allows a single document renderer to render for multiple types of OA documents; each with a different template name.
+
+### 2. name
+
+The `name` key is a compulsory key to describe the type of OA document. In this case, we are creating a `OpenAttestation Tutorial Certificate of Completion`.
+
+### 3. recipient
+
+OA documents do not have strict data structure and allows issuers of documents to define their own data schema. The `recipient` object is a user-defined object that describes who the certificate is conferred to. In this case, you may replace `John Doe` with your name.
+
+In the next section, you will learn more about the OA document schema and how you may defined your own data structure, but for now, we will stick to this simple document data.
+
+### Registering the COC template
+
+`src/templates/index.tsx` is a directory of all the templates available in this renderer.
+
+To register a new template, simply add it as a key to the `registry` constant. Take note that the key will be the `$template.name` value defined in the document data.
+
+Replace `src/templates/index.tsx` with the following code to add the new `COC` template:
+
+```js
+import { TemplateRegistry } from "@govtechsg/decentralized-renderer-react-components";
+import { templates } from "./customTemplate";
+
+export const registry: TemplateRegistry<any> = {
+  default: templates,
+  COC: templates
```

Comment:
It makes me wonder if this is case sensitive, or not, should we add a note on it ?